src/game/client/components/chat.cpp

```diff
@@ -196,8 +197,31 @@ void CChat::ClearChatBuffer()
 
 bool CChat::OnInput(IInput::CEvent Event)
 {
-	if(m_Mode == CHAT_NONE || Client()->State() != Client()->STATE_ONLINE)
+	if(Client()->State() != Client()->STATE_ONLINE)
 		return false;
+	if(m_Mode == CHAT_NONE)
+		return false;
+
+	// chat history scrolling
+	if(m_Show && Event.m_Flags&IInput::FLAG_PRESS && (Event.m_Key == KEY_PAGEUP || Event.m_Key == KEY_PAGEDOWN))
+	{
+		if(Event.m_Flags&IInput::FLAG_PRESS)
+		{
+			if(Event.m_Key == KEY_PAGEUP)
+			{
+				++m_BacklogPage;
+				if(m_BacklogPage >= MAX_CHAT_PAGES) // will be further capped during rendering
+					m_BacklogPage = MAX_CHAT_PAGES-1;
+			}
+			else if(Event.m_Key == KEY_PAGEDOWN)
+			{
+				--m_BacklogPage;
+				if(m_BacklogPage < 0)
+					m_BacklogPage = 0;
+			}
+		}
+		return true;
```

Comment:
Could check if `m_Mode == CHAT_NONE` is true and then return false, then ingame binds should work.

Reply:
Done

Reply:
@oy Hmm, so I think what you are thinking of is 
```cpp
	// chat history scrolling
	if(m_Show && Event.m_Flags&IInput::FLAG_PRESS && (Event.m_Key == KEY_PAGEUP || Event.m_Key == KEY_PAGEDOWN))
	{
		if(Event.m_Key == KEY_PAGEUP)
		{
			++m_BacklogPage;
			if(m_BacklogPage >= MAX_CHAT_PAGES) // will be further capped during rendering
				m_BacklogPage = MAX_CHAT_PAGES-1;
		}
		if(Event.m_Key == KEY_PAGEDOWN)
		{
			--m_BacklogPage;
			if(m_BacklogPage < 0)
				m_BacklogPage = 0;
		}
		return (m_Mode != CHAT_NONE); // trigger binds if not chatting
	}
```

I am not sure if it's such a good idea to trigger pageup/pagedown binds when they are already hardcoded and used for the chat pages though?